### PR TITLE
feat(Core/Command): server debug

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
-  add command `server debug` most of the code is from TrinityCore
-  add GitRevision::GetCMakeVersion & GitRevision::GetHostOSVersion from TrinityCore
- add MySQL::GetLibraryVersion from TrinityCore
- add World::LoadDBRevision to load the exact database version (Requires https://github.com/azerothcore/azerothcore-wotlk/pull/6006)

## Tests Performed:

- Builds
- tested on console
image

## How to Test the Changes:

1. .server debug

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/authserver/Main.cpp [hunk 1]
```diff
@@ -100,7 +100,8 @@ extern int main(int argc, char** argv)
         {
             LOG_INFO("server.authserver", "> Using configuration file       %s.", sConfigMgr->GetFilename().c_str());
             LOG_INFO("server.authserver", "> Using SSL version:             %s (library: %s)", OPENSSL_VERSION_TEXT, SSLeay_version(SSLEAY_VERSION));
-            LOG_INFO("server.authserver", "> Using ACE version:             %s\n", ACE_VERSION);
+            LOG_INFO("server.authserver", "> Using ACE version:             %s", ACE_VERSION);
+            LOG_INFO("server.authserver", "> Using MySQL version:           %u\n", MySQL::GetLibraryVersion());
```
Comment (Winfidonarleyan) [inline src/server/authserver/Main.cpp:104]: no need this here

**Comment by FrancescoBorzi** [comment]
build failing

**Comment by Winfidonarleyan** [comment]
```cpp
void LoadDBRevision()
{
    QueryResult result = WorldDatabase.Query("DESCRIBE `version_db_world`");
    if (!result)
    {
        LOG_ERROR("sql.sql", "> Unkown World Database Revision");
        return;
    }

    std::string dbWorldRevision;

    do
    {
        std::string revision = result->Fetch()->GetString();

        if (revision == "sql_rev" || revision == "required_rev")
            continue;

        dbWorldRevision = revision;
        break;
    } while (result->NextRow());

    LOG_INFO("server.loading", "> Revision DB world:     %s", dbWorldRevision.c_str());
    LOG_INFO("server.loading", "");
}
```

**Comment by Kitzunu** [comment]
mmm sure, but not the best.

I want the date as it will be much easier to see if someone is up-to-date or not

**Comment by Kitzunu** [comment]
Now this PR only requires https://github.com/azerothcore/azerothcore-wotlk/pull/6006

**Comment by Kitzunu** [comment]
Okay now it is fully finished just waiting for the other PR to be merged.

Image updated as well for final output =)

**Review by Winfidonarleyan — APPROVED** [review]

**Comment by Winfidonarleyan** [comment]
```
Run source ./apps/ci/ci-worldserver-dry-run.sh
Cloning into './env/dist/data'...
SQL: SELECT date FROM version_db_world ORDER BY date DESC LIMIT 1
ERROR: [1054] Unknown column 'date' in 'field list'
Your database structure is not up to date. Please make sure you've executed all queries in the sql/updates folders.
timeout: the monitored command dumped core
./apps/ci/ci-worldserver-dry-run.sh: line 11: 17507 Aborted                 timeout 5m ./worldserver --dry-run
Error: Process completed with exit code 134.
```

**Comment by Kitzunu** [comment]
yaya that will be fixed now :D

**Pull request merged**